projects/vdk-plugins/vdk-jobs-troubleshooting/src/vdk/plugin/jobs_troubleshoot/troubleshoot_utilities/healthcheck_server.py

```diff
@@ -21,25 +22,50 @@ def __init__(self, port: int, handler: Any = None):
             port (int): The port number on which the server will listen for requests.
             handler (Any, optional): The request handler class. Defaults to SimpleHTTPRequestHandler.
         """
-        if handler:
-            self._server = HTTPServer(("", port), handler)
-            self._thread = Thread(target=self._server.serve_forever)
-            log.error(f"Troubleshooting utility server started on port {port}.")
-        else:
-            log.error(
-                "Troubleshooting utility handler not specified. Will not start the server."
+        try:
+            if handler:
+                port = self.find_open_port(port)
+                self._server = HTTPServer(("", port), handler)
+                self._thread = Thread(target=self._server.serve_forever)
+                log.info(f"Troubleshooting utility server started on port {port}.")
+            else:
+                log.error(
+                    "Troubleshooting utility handler not specified. Will not start the server."
+                )
+        except Exception as e:
+            logging.error(
```

Comment:
Why not `logging.warning()`? This way the full stacktrace would be added to the message, and the exc_info would not be needed?

Reply:
Do you mean logging.exception()? As far as I can tell to get the exception in the when calling warning() I still have to pass the exc_info paramter?
Anyway... I prefer to keep it like that if you don't mind.

Reply:
I meant `logging.warning()` (according to the documentation, https://docs.python.org/3/library/logging.html#logging.Logger.warning, it behaves like debug which means that it includes the stacktrace by default), but I am ok to leave like that.